hack/cloud-image-uploader/azure.go

```diff
@@ -0,0 +1,420 @@
+	return g.Wait()
+
+}
+
+func setDefaultAzureCreds() error {
+	subscriptionID = os.Getenv("AZURE_SUBSCRIPTION_ID")
+	if len(subscriptionID) == 0 {
+		log.Fatalln("AZURE_SUBSCRIPTION_ID is not set.")
+	}
+
+	authFromEnvironment, err := auth.NewAuthorizerFromEnvironment()
+	if err != nil {
+		return err
+	}
+	authorizer = authFromEnvironment
+
+	//Create a new instance of the DefaultAzureCredential
+	cred, err = azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return err
+	}
+
+	//Initialize the Storage Accounts Client
+	storageClientFactory, err = armstorage.NewClientFactory(subscriptionID, cred, nil)
+	if err != nil {
+		return err
+	}
+	accountsClient = storageClientFactory.NewAccountsClient()
+	providersClient = resources.NewProvidersClient(subscriptionID)
+	providersClient.Authorizer = authorizer
+
+	return nil
+}
+
+func (azu *AzureUploader) createAzureDisk(ctx context.Context, templateFile string, arch string) error {
+
+	diskParameters := map[string]interface{}{
+		"disk_name": map[string]string{
+			"value": "talos-" + arch + "-" + azu.Options.Tag,
+		},
+		"storage_account": map[string]string{
+			"value": storageAccount,
+		},
+		"vhd_name": map[string]string{
+			"value": "talos-" + arch + "-" + azu.Options.Tag + ".vhd",
+		},
+		"region": map[string]string{
+			"value": defaultRegion,
+		},
+		"architecture": map[string]string{
+			"value": azureArchitectures[arch],
+		},
+	}
+
+	deploymentName := "disk-talos-" + arch + "-" + azu.Options.Tag
+
+	err := deployResourceFromTemplate(ctx, subscriptionID, templateFile, diskParameters, deploymentName)
+	if err != nil {
+		fmt.Printf("Error: %v\n", err)
+	}
+	return err
+}
+
+func (azu *AzureUploader) createAzureImageVersion(ctx context.Context, templateFile string, arch string) error {
+
+	var g *errgroup.Group
+
+	g, ctx = errgroup.WithContext(ctx)
+
+	targetRegions := make([]TargetRegion, 0, len(locations))
+
+	for _, region := range locations {
+		targetRegions = append(targetRegions, TargetRegion{
+			Name:                 region.Name,
+			RegionalReplicaCount: 1,
+			StorageAccountType:   "Standard_LRS",
+		})
+	}
+
+	versionParameters := map[string]interface{}{
+		"disk_name": map[string]string{
+			"value": "talos-" + arch + "-" + azu.Options.Tag,
+		},
+		"image_version": map[string]string{
+			"value": azu.Options.Tag,
+		},
+		"gallery_name": map[string]string{
+			"value": galleryName,
+		},
+		"definition_name": map[string]string{
+			"value": "talos-" + azureArchitectures[arch],
+		},
+		"region": map[string]string{
+			"value": defaultRegion,
+		},
+		"resourceGroupName": map[string]string{
+			"value": resourceGroupName,
+		},
+		"targetRegions": map[string]interface{}{
+			"value": targetRegions,
+		},
+	}
+
+	deploymentName := "imgversion-talos-" + arch + "-" + azu.Options.Tag
+
+	err := deployResourceFromTemplate(ctx, subscriptionID, templateFile, versionParameters, deploymentName)
+	if err != nil {
+		fmt.Printf("Error: %v\n", err)
+		return err
```

Comment:
better to wrap an error here, 
```suggestion
		return fmt.Errorf("error creating image version: %w", err)
```